kolibri/logger/migrations/0004_tidy_progress_range.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.10 on 2018-03-28 07:50
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def tidy_progress_range(apps, schema_editor):
+    """
+    Tidies progress ranges because a bug had caused them to go out of range
+    """
+    ContentSessionLog = apps.get_model('logger', 'ContentSessionLog')
+    ContentSummaryLog = apps.get_model('logger', 'ContentSummaryLog')
+
+    # Not knowing how floating points will behave in the local database,
+    # 1.0 might become bigger than 1.0!!
+    ContentSessionLog.objects.filter(progress__lt=0).update(progress=0.0)
+    ContentSessionLog.objects.filter(progress__gt=1).update(progress=1.0)
+    ContentSummaryLog.objects.filter(progress__lt=0).update(progress=0.0)
+    ContentSummaryLog.objects.filter(progress__gt=1).update(progress=1.0)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('logger', '0003_auto_20170531_1140'),
+    ]
+
+    operations = [
+        migrations.RunPython(tidy_progress_range),
```

Comment:
This seems like a risky thing to do without understanding _why_ values are outside `[0, 1]`. If there is invalid data in the DB, my sense is that we should either leave it alone, or understand how it got there and try to fix it

Reply:
Agreed. And the rounding fixes in this PR don't address the root issue; note that some values are as high as 2.7:

https://user-images.githubusercontent.com/5150483/35353927-881d1fbc-00fd-11e8-85ab-cc4434fd5af4.png

For the same reason, I worry about adding the validation logic, since we *know* it's likely to cause saving errors.